### PR TITLE
Additional tests for task list Nunjucks macro, allow attributes on tags and tidy up

### LINKED ISSUE #3828: Improve template test coverage for the task list component
## What

Add additional template tests that cover the various options that can be passed to the macro.

## Why

Improving the coverage will improve our confidence that the component can be used without error – for example, that adding new classes or attributes works as expected.

As a side effect, the new examples that get added as part of writing the tests improve the coverage of the macro in our test fixtures which other teams can use to ensure their ports of GOV.UK Frontend output the same HTML.

As discussed with the original contributor on Slack, they're happy for us to take on the work of adding additional tests on their behalf.

## Who needs to work on this

Developers

## Who needs to review this

Developers

## Done when

- [ ] Tests have been added that cover all of the flows of execution through the Nunjucks template

### PR DESCRIPTION
**Pull request by 36degrees**
Add additional template tests that cover the various options that can be passed to the macro.

See individual commits for details.

Closes #3828.

### CODE REVIEW

File: packages/govuk-frontend/src/govuk/components/task-list/task-list.yaml [hunk 1]
```diff
@@ -41,21 +41,9 @@ params:
...
-                description: Text to use within the tag. If `html` is provided, the `text` argument will be ignored.
-              - name: html
-                type: string
-                required: true
-                description: HTML to use within the tag. If `html` is provided, the `text` argument will be ignored.
-              - name: classes
-                type: string
-                required: false
-                description: Classes to add to the tag.
+            description: Options for the tag component, if the status should use a tag.
```
Comment (colinrotherham) [inline packages/govuk-frontend/src/govuk/components/task-list/task-list.yaml:46]: Nice spot. Do you mind adding this to the macro options style guide Trello?

I've only checked briefly, but other `required: false` components don't qualify if they should be used

```suggestion
            description: Options for the tag component.
```

File: packages/govuk-frontend/src/govuk/components/task-list/template.njk [hunk 2]
```diff
@@ -1,31 +1,30 @@
...
       </div>
-      <div class="govuk-task-list__status {{ item.status.classes }}" id="{{ statusId }}">
+      <div class="govuk-task-list__status {%- if item.status.classes %} {{ item.status.classes }}{% endif %}" id="{{ statusId }}">
         {% if item.status.tag %}
           {{ govukTag({
             text: item.status.tag.text,
             html: item.status.tag.html,
-            classes: item.status.tag.classes
+            classes: item.status.tag.classes,
+            attributes: item.status.tag.attributes
```
Comment (colinrotherham) [inline packages/govuk-frontend/src/govuk/components/task-list/template.njk:27]: Other nested components usually override `id` or `for` params

But we don't here. Should we just pass all the options in one?

```mjs
{{ govukTag(item.status.tag) }}
```

**Review by colinrotherham — COMMENTED** [review]
Looks brilliant. Every single commit

Mind if we update the PR title + description to say we've added fixes and features rather than just better test coverage?

File: packages/govuk-frontend/src/govuk/components/task-list/template.njk (hunk 2 above)
Comment (36degrees) [reply packages/govuk-frontend/src/govuk/components/task-list/template.njk:27]: That is a great shout 👍🏻

File: packages/govuk-frontend/src/govuk/components/task-list/task-list.yaml (hunk 1 above)
Comment (36degrees) [reply packages/govuk-frontend/src/govuk/components/task-list/task-list.yaml]: > Nice spot. Do you mind adding this to the macro options style guide Trello?

https://trello.com/c/smPF3ePN/18-nested-components-components-used-by-other-components-should-not-have-their-params-listed-in-the-top-level-component-instead-sho

**Review by colinrotherham — APPROVED** [review]

**Pull request merged**